prepline_general/api/general.py

```diff
@@ -58,6 +62,84 @@ def is_expected_response_type(media_type, response_type):
     os.environ["UNSTRUCTURED_ALLOWED_MIMETYPES"] = DEFAULT_MIMETYPES
 
 
+def get_pdf_splits(pdf, split_size=1):
+    """
+    Given a pdf (PdfReader) with n pages, split it into pdfs each with split_size # of pages
+    Return the files with their page offset in the form [( BytesIO, int)]
+    """
+    split_pdfs = []
+
+    offset = 0
+
+    while offset < len(pdf.pages):
+        new_pdf = PdfWriter()
+        pdf_buffer = io.BytesIO()
+
+        end = offset + split_size
+        for page in pdf.pages[offset:end]:
+            new_pdf.add_page(page)
+
+        new_pdf.write(pdf_buffer)
+        pdf_buffer.seek(0)
+
+        split_pdfs.append((pdf_buffer, offset))
+        offset += split_size
+
+    return split_pdfs
+
+
+def partition_file_via_api(file_tuple, **kwargs):
+    """
+    Given a file-like, use partition_via_api to retrieve its elements.
+    """
+    # Note (austin) - we'll need to have the Request object here to forward
+    # any important context, e.g. api keys
+    request_url = os.environ.get("UNSTRUCTURED_PARALLEL_MODE_URL")
+
+    if not request_url:
+        raise HTTPException(status_code=500, detail="Parallel mode enabled but no url set!")
+
+    file, page_offset = file_tuple
+
+    try:
+        # Note (austin) - consider using partition_multiple_via_api to cut down on traffic
+        # This would serialize the work that we just tried to split up,
+        # so we first need to parallelize multi file input
+        elements = partition_via_api(file=file, api_url=request_url, **kwargs)
```

Comment:
might want a retry decorator here. ie, retry once on a spurious error

Reply:
That makes sense! I'll make a note for the next pass.